tests/Catalog.FunctionalTests/CatalogApiFixture.cs

```diff
@@ -25,7 +25,7 @@ protected override IHost CreateHost(IHostBuilder builder)
         {
             config.AddInMemoryCollection(new Dictionary<string, string>
             {
-                { $"ConnectionStrings:{Postgres.Resource.Name}", Postgres.Resource.GetConnectionString() },
+                { $"ConnectionStrings:{Postgres.Resource.Name}", Postgres.Resource.ConnectionStringExpression.ValueExpression },
```

Comment:
@mitchdenny and I discussed this part. Worth thinking about what the dev experience for this should be.

Reply:
So for now I think we would need to do sync over async here which is obviously not ideal. Adding @ReubenBond and @davidfowl for their thoughts on how we can stitch the app model together with WAF.

Reply:
Or use @ReubenBond’s new testing support. It would be a good stress test

Reply:
The Aspire.Hosting.Testing package has GetConnectionStringAsync (the underlying Aspire Hosting API is async)

Reply:
Fixed